### DISCUSSION TITLE
Sigmoid instead of ReLu in Binary Classification

### DISCUSSION THREAD

**Discussion by sac303** [mrdbourke/pytorch-deep-learning#858] [Q&A]
I tried to use sigmoid instead of ReLU inside the model, as it also activates the non-linear function. I kept everything the same based on the tutorial. However, my model is not learning anything. When I tried this in the TensorFlow playground, it appeared to work. Could someone explain this?

Code - 

 from torch import nn
 class CircleModelV3(nn.Module):
  def __init__(self):
    super().__init__()

    self.layer_1 = nn.Linear(in_features=2, out_features=10)
    self.layer_2 = nn.Linear(in_features=10, out_features=10)
    self.layer_3 = nn.Linear(in_features=10, out_features = 1)
    self.sigmoid = nn.sigmoid() 
    

  def forward(self,x:torch.tensor):
    return self.layer_3(self.sigmoid(self.layer_2(self.sigmoid(self.layer_1(x)))))
'''

**Comment by LuluW8071** [comment]
### Correct way to use Sigmoid activation module is `nn.Sigmoid` not `nn.sigmoid`
#### Try this:

```python
class CircleModelV3(nn.Module):
    def __init__(self):
        super().__init__()

        self.layer_1 = nn.Linear(in_features=2, out_features=10)
        self.layer_2 = nn.Linear(in_features=10, out_features=10)
        self.layer_3 = nn.Linear(in_features=10, out_features=1)
        # Correct usage of the Sigmoid activation module
        self.sigmoid = nn.Sigmoid()  

    def forward(self, x: torch.tensor):
        return self.layer_3(self.sigmoid(self.layer_2(self.sigmoid(self.layer_1(x)))))
```
